### ISSUE TITLE
Control the precision of units

### ISSUE THREAD

**Issue by arkanoid87** [SciNim/Unchained#35]
Afaik currently only int/float64 units are handled. How difficult would be introducing 32/16 bit variants?

**Comment by Vindaar** [comment]
In theory it wouldn't be difficult, but I'm not sure about the best way to make it a user option. Generally while currently there are some explicit `float` conversions thrown into the code, those could all become `T` conversions, where `T` is some globally defined type that is the internal data representation. The important one is the initial definition of `Unit = distinct float` here https://github.com/SciNim/Unchained/blob/master/src/unchained/core_types.nim#L7) (conversion could also be done via `Unit.distinctBase` now that I think about it).
By default a user definable `T` would be `float`. The question is how to change that without having to hack around in the code? Try to read some file at CT that may contain a type? I don't think there's a way to override a _type_ via a `define`. Only int, str and bool are possible via `-d:foo=bar`.  One alternative would be:

```nim
const Bytes {.intdefine.} = 8
type
  when Bytes == 8: 
    Unit = distinct float64
  elif Bytes == 4:
    ...
```

That would be relatively neat. 

Regarding `int`: Integers are not really supported in any sense. You may just "think" they are supported in the sense that you can use integer literals in some contexts. But that's just the nim compiler auto converting literals.

**Comment by Vindaar** [comment]
> a compile time flag defining float size in bytes defaulting to 8 would be really neat imho

If you're fine with that solution based on byte size of the float type for now (so currently means only supporting `float32` and `float64`), I can implement that tomorrow. At least cuts your RAM usage ~by half (depending on how much other stuff is non float).

**Comment by Vindaar** [comment]
See the comment in the PR. Hope this works for you.

**Comment by Vindaar** [comment]
> Downgrading to v0.3.1 (you forget to tag this version) or v0.3.0 fixes it.

Oh, sorry. Just forgot to push that tag. Can you check the issue was or wasn't present on `v0.3.1`?

**Comment by Vindaar** [comment]
Not a git bisect, but a small example that shows the issue (unless this happens with any usage of `min` for you?)

**Comment by Vindaar** [comment]
Fixed in `v0.3.3`.
